xain_fl/grpc/test_coordinator.py

```diff
@@ -196,3 +234,42 @@ def test_remove_participant():
     coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
 
     assert coordinator.state == coordinator_pb2.State.ROUND
+
+
+def test_number_of_selected_participants():
+    # test that the coordinator needs minimum 3 participants and selects 2 of them
+    coordinator = Coordinator(
+        minimum_participants_in_round=2, fraction_of_participants=0.6
+    )
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant2")
+
+    # the coordinator should not start a round but wait for a third
+    assert coordinator.minimum_connected_participants == 3
+    assert coordinator.state == coordinator_pb2.State.STANDBY
+
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant3")
+
+    # now the coordinator must have started a round and selected 2 participants
+    assert coordinator.state == coordinator_pb2.State.ROUND
+    assert len(coordinator.round.participant_ids) == 2
+
+
+def test_correct_round_advertised_to_participants():
+    # test that only selected participants receive ROUND state and the others STANDBY
+    coordinator = Coordinator(
+        minimum_participants_in_round=1, fraction_of_participants=0.5
+    )
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant2")
+
+    for participant_id in coordinator.participants.ids():
```

Comment:
this test could be clearer (and simpler) if you knew exactly which were selected e.g. by mocking the selection. there are only two participants after all.

Reply:
makes totally sense. test updated, thanks